openpathsampling/experimental/simstore/storage.py

```diff
@@ -587,14 +585,27 @@ def __iter__(self):
 
     def __getitem__(self, item):
         len_self = len(self)
-        if not (-len_self <= item < len_self):
-            raise IndexError("table index out of range")
-
-        if item < 0:
-            item += len(self)
+        if type(item) is slice:  # Slice is not an acceptable base class
+            unpack_return = False
+            items = (i for i in range(*item.indices(len_self)))
+        elif type(item) is int:
+            unpack_return = True
+            if not (-len_self <= item < len_self):
+                raise IndexError("table index out of range")
+
+            if item < 0:
+                item += len(self)
+            items = [item]
+        else:
+            raise TypeError("Only access via slice or int allowed, "
+                            f"got type {type(item).__name__}.")
 
-        row = self.storage.backend.table_get_item(self.table, item)
-        return self.storage.load([row.uuid])[0]
+        rows = [self.storage.backend.table_get_item(self.table, item).uuid
+                for item in items]
+        if unpack_return:
+            return self.storage.load(rows)[0]
+        else:
+            return self.storage.load(rows)
```

Comment:
```suggestion
        uuids = [self.storage.backend.table_get_item(self.table, item).uuid
                 for item in items]
        if unpack_return:
            return self.storage.load(uuids)[0]
        else:
            return self.storage.load(uuids)
```

Rename `rows` => `uuids` for readability. Usually `row` refers to the entire entry in the database (all columns).